test/jdk/java/lang/management/MemoryMXBean/MemoryTest.java

```diff
@@ -46,17 +46,29 @@
  */
 
 /*
- * @test
+ * @test id=Shenandoah
  * @bug     4530538
- * @summary Basic unit test of MemoryMXBean.getMemoryPools() and
- *          MemoryMXBean.getMemoryManager().
- * @requires vm.gc == "Shenandoah"
+ * @summary Shenandoah has a gc mgr bean for cycles and another
+ *          for pauses, they both have one pool.
+ * @requires vm.gc == "Shenandoah" & vm.opt.ShenandoahGCMode != "generational"
  * @author  Mandy Chung
  *
  * @modules jdk.management
  * @run main MemoryTest 2 1
  */
 
+/*
+ * @test id=Genshen
+ * @bug     4530538
```

Comment:
Idle thought: should this stay 4530538 (I suppose!), or should it be 8346051? :-) 
Probably should stay 4530538 because of https://openjdk.org/jtreg/faq.html#how-do-i-run-only-tests-which-were-written-for-a-specific-bugid .